pkg/frontend/querymiddleware/codec.go

```diff
@@ -315,11 +317,16 @@ func (prometheusCodec) EncodeRequest(ctx context.Context, r Request) (*http.Requ
 		Header:     http.Header{},
 	}
 
+	if c.requestProtobufFormat {
```

Comment:
Stylistic but, I'd just switch on the value of `QueryResultPayloadFormat` here to set the accept header instead of storing a `useX`-style boolean in the struct.

Reply:
Makes sense, changed in cd8674b.